packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,231 @@
+                              droppableId="hidden-columns-panel"
+                              data-testid="hidden-columns-panel"
+                              noColumnsText={formatMessage(
+                                messages.noHiddenColumnsToShow
+                              )}
+                              columns={hiddenColumns}
+                              isSearchable={props.areHiddenColumnsSearchable}
+                            />
+                          </Spacings.Stack>
+                        </DroppableContainer>
+                        <DroppableContainer isDragging={isDragging}>
+                          <Spacings.Stack>
+                            <Spacings.Inline scale="xs" alignItems="center">
+                              <EyeIcon size="medium" />
+                              <FieldLabel
+                                title={formatMessage(messages.visibleColumns)}
+                              />
+                            </Spacings.Inline>
+                            <DroppablePanel
+                              droppableId="selected-columns-panel"
+                              data-testid="selected-columns-panel"
+                              noColumnsText={formatMessage(
+                                messages.noSelectedColumnsToShow
+                              )}
+                              columns={props.selectedColumns}
+                              onRemove={props.onUpdateColumns}
+                            />
+                          </Spacings.Stack>
+                        </DroppableContainer>
+                      </Spacings.Inline>
+                    </DragDropContext>
+                  </div>
+                </div>
+              </Spacings.Inset>
+              {(props.secondaryButton || props.primaryButton) && (
+                <Spacings.Inline justifyContent="flex-end">
+                  {props.secondaryButton}
+                  {props.primaryButton}
+                </Spacings.Inline>
+              )}
+            </Spacings.Stack>
+          </Spacings.Stack>
+        </Container>
+      )}
+    </CollapsibleMotion>
+  );
+};
+
+ColumnManager.displayName = 'ColumnManager';
+
+ColumnManager.propTypes = {
+  availableColumns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    })
+  ).isRequired,
+  selectedColumns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    })
+  ).isRequired,
+  onUpdateColumns: PropTypes.func.isRequired,
+
+  areHiddenColumnsSearchable: PropTypes.bool,
+  searchHiddenColumns: requiredIf(
+    PropTypes.func,
+    (props) => props.areHiddenColumnsSearchable
+  ),
+  searchHiddenColumnsPlaceholder: PropTypes.string,
+
+  onClose: PropTypes.func,
+  primaryButton: PropTypes.node,
```

Comment:
I don't see a test for these now them being used anywhere. Is that intended? Furthermore, nodes are anything that can be rendered. So strings, numbers... I don't feel like this is what we mean here if I read things correctly. Do we mean `element`?

Reply:
Actually I've change that in the root component but didn't propagate the change 😜